das/daser.go

```diff
@@ -18,21 +19,37 @@ type DASer struct {
 	da   share.Availability
 	hsub header.Subscriber
 
-	cancel context.CancelFunc
-	done   chan struct{}
+	// getter allows the DASer to fetch a header at a certain height
+	// and blocks until it becomes available.
```

Comment:
@adlerjohn suggested in the past to the DASing and the availability check async. But IMO we could do this blocking  as it's done here first.

Reply:
Yeah that would be a cute optimization, but not strictly necessary.

Reply:
By `available`, it means the node has the header.

Reply:
That the network is at least @ that height

Reply:
> By available, it means the node has the header.

That is really confusing in this context. Esp. as you also ensure availability of the data behind the header later as well.

Reply:
Also: 

> That the network is at least @ that height

Looking at how this method is used, the network will always be past that height (as you only call it after pubsub gives you some larger height).

Reply:
> That is really confusing in this context. Esp. as you also ensure availability of the data behind the header later as well.

Okay this is a good point


> Looking at how this method is used, the network will always be past that height (as you only call it after pubsub gives you some larger height).

Yes this is also true but we need to have stored and processed the EH first to be able to DAS over it since we need the DAH. I can document this.

Reply:
can you check now?

Reply:
> suggested in the past to the DASing and the availability check async
> Yeah that would be a cute optimization, but not strictly necessary.

WDYM? Isn't the DAS of height is the same as the availability check? Yes, I read through the thread and I understand that @renaynay meant a different thing, but I am curious how you misunderstood this and what optimization you meant. I believe we have all such cute optimizations already.

Reply:
I think I understood it as being able to to DAS on multiple blocks simultaneously without blocking, e.g. during initial sync.

Reply:
Yes. We will be able to DAS on blocks during sync, but we can only DAS on headers that are already known to the node, hence why we need the `HeaderGetter`. I'm not sure where the confusion lies anymore.

Reply:
@adlerjohn @liamsi  resolved?